src/Tasks/FileIO/GetFileHash.cs

```diff
@@ -64,31 +60,77 @@ public override bool Execute()
                 return false;
             }
 
+            if (!TryParseHashEncoding(HashEncoding, out var encoding))
+            {
+                Log.LogErrorFromResources("FileHash.UnrecognizedHashEncoding", HashEncoding);
+                return false;
+            }
+
             foreach (var file in Files)
             {
                 if (!File.Exists(file.ItemSpec))
                 {
                     Log.LogErrorFromResources("FileHash.FileNotFound", file.ItemSpec);
-                    continue;
                 }
+            }
+
+            if (Log.HasLoggedErrors)
+            {
+                return false;
+            }
 
+            foreach (var file in Files)
+            {
                 var hash = ComputeHash(Algorithm, file.ItemSpec);
                 file.SetMetadata("FileHashAlgoritm", Algorithm);
-                file.SetMetadata(MetadataName, ConversionUtilities.ConvertByteArrayToHex(hash));
-                file.SetMetadata(MetadataNameBase64, Convert.ToBase64String(hash));
+                file.SetMetadata(MetadataName, EncodeHash(encoding, hash));
             }
 
             Items = Files;
 
             if (Files.Length == 1)
             {
                 Hash = Files[0].GetMetadata(MetadataName);
-                HashBase64 = Files[0].GetMetadata(MetadataNameBase64);
             }
 
             return !Log.HasLoggedErrors;
         }
 
+        internal string EncodeHash(HashEncoding encoding, byte[] hash)
+        {
+            switch (encoding)
+            {
+                case Tasks.HashEncoding.Hex:
+                    return ConversionUtilities.ConvertByteArrayToHex(hash);
+                case Tasks.HashEncoding.Base64:
+                    return Convert.ToBase64String(hash);
+                default:
+                    throw new NotImplementedException();
+            }
+        }
+
+        internal static bool TryParseHashEncoding(string value, out HashEncoding encoding)
```

Comment:
Is there a particular reason you are using switch with all the hardcoded text strings here rather than just `return Enum.TryParse<HashEncoding>(value, true, out encoding);`

I am thinking about ensuring that if new encoding methods are added in the future, that it doesn't require changes in lots of places.

Reply:
No particular reason. Enum.TryParse is better so switched to that.